src/cascadia/TerminalApp/App.base.h

```diff
@@ -9,30 +9,23 @@ namespace winrt::TerminalApp::implementation
 
         IXamlType GetXamlType(::winrt::Windows::UI::Xaml::Interop::TypeName const& type)
         {
-            return AppProvider()->GetXamlType(type);
+            return AppProvider().GetXamlType(type);
         }
 
         IXamlType GetXamlType(::winrt::hstring const& fullName)
         {
-            return AppProvider()->GetXamlType(fullName);
+            return AppProvider().GetXamlType(fullName);
         }
 
         ::winrt::com_array<::winrt::Windows::UI::Xaml::Markup::XmlnsDefinition> GetXmlnsDefinitions()
         {
-            return AppProvider()->GetXmlnsDefinitions();
+            return AppProvider().GetXmlnsDefinitions();
         }
 
     private:
         bool _contentLoaded{ false };
-        std::shared_ptr<XamlMetaDataProvider> _appProvider;
-        std::shared_ptr<XamlMetaDataProvider> AppProvider()
-        {
-            if (!_appProvider)
-            {
-                _appProvider = std::make_shared<XamlMetaDataProvider>();
-            }
-            return _appProvider;
-        }
+        winrt::TerminalApp::XamlMetaDataProvider _appProvider;
+        winrt::TerminalApp::XamlMetaDataProvider AppProvider() { return _appProvider; };

```

Comment:
So, the only thing we got from having it implemented like this was that it was lazy-created. Two questions:

1. If we're not lazy-creating it anymore, we can drop the function and just use `_appProvider` directly.
2. Does this result in a behavior change that might impact reliability? I'm moderately worried about that, but I bet that it was getting hit instantly on startup anyway...

Reply:
I'm not the expert on this, if you and others say we can just use _appProvider directly, I will gladly make that change.

Reply:
Sure, let's do it. If we're gonna get rid of lazy-loading, let's go all the way.

The downside of the function approach as it stands (and the only reason i didn't say "let's just merge this") is that it causes a reference count change every time you call AppProvider(), because it returns a copy of the smart pointer. :smile:

Reply:
Fair enough, since it doesn't serve a purpose anymore, I removed the getter now :)